ibis/expr/api.py

```diff
@@ -200,6 +202,57 @@ def time(value):
     return ir.TimeScalar(ir.literal(value).op())
 
 
+def interval(value=None, years=None, months=None, weeks=None, days=None,
+             hours=None, minutes=None, seconds=None):
+    """
+    Returns an interval literal
+
+    Parameters
+    ----------
+    value : int or datetime.timedelta, default None
+    years : int, default None
+    months : int, default None
+    days : int, default None
+    weeks : int, default None
+    hours : int, default None
+    minutes : int, default None
+    seconds : int, default None
+
+    Returns
+    --------
+    result : IntervalScalar
+    """
+    # TODO: consider (value, unit) arguments instead of kwargs
+
+    if value is not None:
+        type = dt.Interval('s')
+        if isinstance(value, datetime.timedelta):
+            value = int(value.total_seconds())
+        elif not isinstance(value, six.integer_types):
+            raise ValueError('Interval value must be an integer')
+        return ir.IntervalScalar(ir.literal(value, type=type).op())
+
+    kwds = [
+        ('Y', years),
+        ('M', months),
+        ('w', weeks),
+        ('d', days),
+        ('h', hours),
+        ('m', minutes),
+        ('s', seconds)
+    ]
+    defined_units = [(k, v) for k, v in kwds if v is not None]
+
+    if len(defined_units) > 1:
+        raise ValueError('Only one arg can be specified')
+    elif len(defined_units) < 1:
+        raise ValueError('At least one of the arguments must be specified')
```

Comment:
Probably just `len(defined_units) != 1`, with a message saying `Exactly one argument is required`.

Is there any reason why we need to enforce this? Postgres, for example, allows multiple values to the interval constructor.

Reply:
Postgres is more flexible thank [others](https://github.com/ibis-project/ibis/issues/1233)

Reply:
Fair enough, just wondering.